text/chipseq_workflow.md

```diff
@@ -0,0 +1,327 @@
+  md5sum -c gencode.v31.annotation.gtf.gz.md5
+  # > gencode.v31.annotation.gtf.gz: OK
+
+  gunzip gencode.v31.annotation.gtf.gz
+  echo "4e22351ae216e72aa57cd6d6011960f8  gencode.v31.annotation.gtf" > gencode.v31.annotation.gtf.md5
+  md5sum -c gencode.v31.annotation.gtf.md5
+  # > gencode.v31.annotation.gtf: OK
+  ```
+
+- A list of excluded regions are obtained from the Boyle lab. This list is based on ENCODE data and a comprehensive set of anomalous, unstructured, or high-signal regions in next generation sequencing data.
+  ```bash
+  wget -O hg38.excludelist.bed.gz https://github.com/Boyle-Lab/Blacklist/raw/master/lists/hg38-blacklist.v2.bed.gz
+  echo "83fe6bf8187a64dee8079b80f75ba289  hg38.excludelist.bed.gz" > hg38.excludelist.bed.gz.md5
+  md5sum -c hg38.excludelist.bed.gz.md5
+  # > hg38.excludelist.bed.gz: OK
+
+  gunzip hg38.excludelist.bed.gz
+  echo "766848c2a632d1b9dd12fac98ce02bd3  hg38.excludelist.bed" > hg38.excludelist.bed.md5
+  md5sum -c hg38.excludelist.bed.md5
+  # > hg38.excludelist.bed: OK
+  ```
+
+## Workflow
+
+Here are the resulting steps in the ChIP-Seq Workflow pipeline. There might be slight alterations in the actual implementation, which can be found in [the St. Jude Cloud workflows repository](https://github.com/stjudecloud/workflows/blob/master/workflows/chipseq/chipseq-standard.wdl).
+
+1. Run `picard ValidateSam` on the incoming BAM to ensure that it is well-formed enough to convert back to FastQ.
+
+   ```bash
+   picard ValidateSamFile I=$INPUT_BAM \                # Input BAM.
+                     IGNORE=INVALID_PLATFORM_VALUE \    # Validations to ignore.
+                     IGNORE=MISSING_PLATFORM_VALUE
+   ```
+
+2. Run `samtools quickcheck` on incoming BAM to ensure it is properly formatted.
+
+   ```bash
+   samtools quickcheck $INPUT_BAM
+   ```
+
+3. Split BAM file into multiple BAMs on the different read groups using `samtools split`. See [the samtools documentation](http://www.htslib.org/doc/samtools.html) for more information.
+
+   ```bash
+   samtools split -u $UNACCOUNTED_BAM_NAME \ # Reads that do not belong to a read group or the read group is unrecognized go here.
+                  -f '%*_%!.%.'              # Format of output BAM file names.
+   ```
+
+   If the BAM has unaccounted reads, those reads will need to be removed and the samtools split step will need to be rerun.
+
+4. Run Picard `SamToFastq` on each of the BAMs generated in the previous step.
+
+   ```bash
+      if [ "~{paired_end}" == 'true' ]
+      then
+         picard SamToFastq \
+             INPUT=$INPUT_BAM \
+             FASTQ=$FASTQ_R1 \
+             SECOND_END_FASTQ=$FASTQ_R2 \
+             RE_REVERSE=true \
+             VALIDATION_STRINGENCY=SILENT
+      else
+         picard SamToFastq \
+             INPUT=$INPUT_BAM \
+             FASTQ=$FASTQ_R1 \
+             RE_REVERSE=true \
+             VALIDATION_STRINGENCY=SILENT
+      fi
+   ```
```

Comment:
`picard SamToFastq` is deprecated on `main` of workflows. But maybe this step is done in SEAseq? If this is meant to be handled by `workflows/` code, it should be changed to use `samtools.collate_to_fastq` (which is admittedly a messier code snippet to put in this document).